### PR TITLE
Archive v1 contracts if blockheight is passed V2 require height

### PR DESCRIPTION
**Pull request by peterjan**
NOTE: can't write an integration test for it because the host doesn't allow renewing contracts after the V2 require height, this means that the contractor will always archive contracts because they are expired and we can't mine passed the V2 require height 

Fixes https://github.com/SiaFoundation/renterd/issues/1545

### CODE REVIEW

File: bus/routes.go [hunk 1]
```diff
@@ -166,9 +166,7 @@ func (b *Bus) consensusStateHandler(jc jape.Context) {
 }
 
 func (b *Bus) consensusNetworkHandler(jc jape.Context) {
-	jc.Encode(api.ConsensusNetwork{
-		Name: b.cm.TipState().Network.Name,
-	})
+	jc.Encode(*b.cm.TipState().Network)
```
Comment (ChrisSchinnerl) [inline bus/routes.go:169]: Sneaky stuff. This is fine but might require a UI update. cc @alexfreska

File: stores/sql/sqlite/migrations/main/schema.sql [hunk 2]
```diff
@@ -21,6 +21,7 @@ CREATE INDEX `idx_contracts_start_height` ON `contracts`(`start_height`);
 CREATE INDEX `idx_contracts_state` ON `contracts`(`state`);
 CREATE INDEX `idx_contracts_window_end` ON `contracts`(`window_end`);
 CREATE INDEX `idx_contracts_window_start` ON `contracts`(`window_start`);
+CREATE INDEX `idx_contracts_v2` ON `contracts`(`v2`);
```
Comment (ChrisSchinnerl) [inline stores/sql/sqlite/migrations/main/schema.sql:24]: I don't think we ever query by the version so we can probably avoid the index.

File: stores/sql/types.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ type (
 	FileContractID  types.FileContractID
 	Hash256         types.Hash256
 	MerkleProof     struct{ Hashes []types.Hash256 }
+	NullableInt64   int64
```
Comment (ChrisSchinnerl) [inline stores/sql/types.go:40]: Didn't we just get rid of this in a different PR and replaced it with just using `*int64`?

File: stores/sql/rows.go
Comment (ChrisSchinnerl) [inline stores/sql/rows.go:1]: I think we got a conflict in this file now

File: bus/routes.go (hunk 1 above)
Comment (alexfreska) [reply bus/routes.go:169]: @ChrisSchinnerl actually the UI does not use this endpoint / its not in our SDKs, so no breaking change but I will add the endpoint to our SDKs.

File: stores/sql/sqlite/migrations/main/schema.sql (hunk 2 above)
Comment (peterjan) [reply stores/sql/sqlite/migrations/main/schema.sql:24]: Feels (to me) like something we should index regardless but I'll remove it.

**Review by ChrisSchinnerl — APPROVED** [review]
Lgtm. Should be good to go once the migrations were tested

**Comment by peterjan** [comment]
> Lgtm. Should be good to go once the migrations were tested

Verified on MySQL instance and run the query on both types of instances as well as verified the migration identifier is ok. Don't immediately have a SQLite instance handy but should be good to go.

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**